src/nwb_benchmarks/core/_streaming.py

```diff
@@ -179,3 +181,39 @@ def read_hdf5_nwbfile_ros3(s3_url: str, retry: bool = True) -> Tuple[pynwb.NWBFi
         retries = None
         nwbfile = io.read()
     return (nwbfile, io, retries)
+
+
+def read_zarr(s3_url: str, force_no_consolidated_metadata=False) -> zarr.Group:
+    """
+    Open a Zarr file from an S3 URL using the built-in fsspec support in Zarr.
+
+    Returns
+    -------
+    file : zarr.Group
+       The zarr.Group object representing the opened file
+    """
+    if force_no_consolidated_metadata:
+        zarrfile = zarr.open(store=s3_url, mode="r", storage_options=dict(anon=True))
+    else:
+        try:
+            zarrfile = zarr.open_consolidated(store=s3_url, mode="r", storage_options=dict(anon=True))
+        except KeyError:
+            zarrfile = zarr.open(store=s3_url, mode="r", storage_options=dict(anon=True))
```

Comment:
Is there no logic we can ask of the Zarr API as to whether there is consolidated metadata available? And if so, I'd adjust the argument here to be `use_consolidated_metadata` which fails with an error if there isn't any

Otherwise, the 'safe' way of just attempting to force it if possible could lead to unclear results in the testing suite since we didn't get feedback as to which route through the logic it actually took on a test case

Reply:
Good point. I have not found anything in Zarr that would allow us to check. The only other check would be to open the file with zarr.open and then check if the .zmetadata attribute exists. Either way there is a check on the remote filesystem needed to see if the .zmetadata file exists. The same logic is being used in hdmf-zarr right now as well. 

That being said, for the benchmarks we should know whether a file does or does not have consolidated metadata (in fact, we probably want to only test with Zarr files that do have it). So maybe we can just always explicitly control in the benchmarks settings whether to use consolidated metadata or not. I.e,. change `force_no_consolidated_metadata` to `open_with_consolidated_metadata` and change this function to:

```
if open_with_consolidated_metadata:
     zarrfile = zarr.open_consolidated(store=s3_url, mode="r", storage_options=dict(anon=True))
else"
        zarrfile = zarr.open(store=s3_url, mode="r", storage_options=dict(anon=True))
```

In this case, if a file does not have consolidated metadata, a test case might fail, but that might be better than benchmarking the wrong thing.

Reply:
I've updated the `read_zarr` method accordingly. If we use a Zarr file without consolidated metadata in benchmarks that assume consolidated metadata, then those benchmarks will fail. But this seems preferable to benchmarking the wrong behavior.